### ISSUE TITLE
How build goyacc tool in GO

### ISSUE THREAD

**Issue by questionpython** [cznic/goyacc#23]
hi
How build goyacc tool in GO?
$ make
--------

>go fmt
go test
PASS
ok  	github.com/cznic/goyacc	0.002s
go install
go tool vet -printfuncs "Log:0,Logf:1" *.go
golint .
make todo
make[1]: Entering directory '/home/guest/go/src/github.com/cznic/goyacc'
main.go:554:	_ = yyerrok
main.go:706:	_ = yypt // guard against "declared and not used"
main.go:791:	_ = oNoLines //TODO Ignored for now
all_test.go:37:func TODO(...interface{}) string {
all_test.go:39:	return fmt.Sprintf("TODO: %s:%d:\n", path.Base(fn), fl)
main.go:791:	_ = oNoLines //TODO Ignored for now
make[1]: Leaving directory '/home/guest/go/src/github.com/cznic/goyacc'

**Comment by cznic** [comment]
I'm not sure if I understand the problem. Anyway, please try issuing

```bash
$ go get -u github.com/cznic/goyacc
```

If that does not work for you, please report back, preferably including the output of

```bash
$ go get -u -v github.com/cznic/goyacc
```

or any other diagnostics you can provide, thank you.

**Comment by questionpython** [comment]
i first delete `/home/guest/go/src/github.com/cznic/goyacc` folder,

$ go get -u -v github.com/cznic/goyacc
---------------
>github.com/cznic/goyacc (download)
github.com/cznic/mathutil (download)
github.com/cznic/parser (download)
github.com/cznic/golex (download)
github.com/cznic/strutil (download)
github.com/cznic/sortutil (download)
github.com/cznic/y (download)
github.com/cznic/golex/lex
github.com/cznic/mathutil
github.com/cznic/strutil
github.com/cznic/sortutil
github.com/cznic/parser/yacc
github.com/cznic/y
github.com/cznic/goyacc

**Comment by cznic** [comment]
Well, the output seems to indicate the installation went smoothly with no problems. You should be now able to find the `goyacc` binary in `$GOPATH/bin`, or alternatively, in `$GOBIN`, if set. Whichever that location is, it has to be listed in your `$PATH` if you now want to invoke the command `goyacc` from any working directory.

**Comment by cznic** [comment]
Here it seems to work as well:

```bash
jnml@4670:~$ ls -a $(which goyacc)
/home/jnml/bin/goyacc
jnml@4670:~$ rm /home/jnml/bin/goyacc
jnml@4670:~$ go get -u -v github.com/cznic/goyacc
github.com/cznic/goyacc (download)
github.com/cznic/mathutil (download)
github.com/cznic/parser (download)
github.com/cznic/golex (download)
github.com/cznic/strutil (download)
github.com/cznic/sortutil (download)
github.com/cznic/y (download)
github.com/cznic/goyacc
jnml@4670:~$ ls -la $(which goyacc)
-rwxrwxr-x 1 jnml jnml 3397456 bře  9 20:33 /home/jnml/bin/goyacc
jnml@4670:~$ 
```

**Comment by questionpython** [comment]
$ ls -a $(which goyacc)
/home/guest/go/bin/goyacc

**Comment by cznic** [comment]
That seems okay.

**Comment by questionpython** [comment]
$ make
--------

go fmt
go test
PASS
ok  	github.com/cznic-/goyacc	0.002s
go install
go tool vet -printfuncs "Log:0,Logf:1" *.go
golint .
make todo
make[1]: Entering directory '/home/guest/go/src/github.com/cznic-/goyacc'
main.go:554:	_ = yyerrok
main.go:706:	_ = yypt // guard against "declared and not used"
main.go:791:	_ = oNoLines //TODO Ignored for now
all_test.go:37:func TODO(...interface{}) string {
all_test.go:39:	return fmt.Sprintf("TODO: %s:%d:\n", path.Base(fn), fl)
main.go:791:	_ = oNoLines //TODO Ignored for now
make[1]: Leaving directory '/home/guest/go/src/github.com/cznic-/goyacc'

**Comment by cznic** [comment]
> you sure then test goyacc on GO 1.8?

Yes, you should now have the `goyacc` binary built by you current Go toolchain. For the record, `goyacc` binary built from a recent source version, but with the older version of the Go toolchain should work just fine. 

You don't need to invoke `make`. `go get -u <import-path>` or `go install <import-path>` already produces/updates the binary. `make` is a `goyacc`'s developer aid only, not required for end-users ;-) 

Anyway running `make` also installs the `goyacc` command.

**Comment by questionpython** [comment]
$ go install github.com/cznic/goyacc
-------
result of this is empty.!

**Comment by cznic** [comment]
> result of this is empty.!

Plase specify what is meant by *empty*. It looks like you have the correct `goyacc` binary installed as seen from the output of `goyacc -h`. I don't know what the problem is, before I thought it was about *having* the `goyacc` binary. Is there some other problem I'm not aware of?

I guess there might be a problem of communicating in English, maybe Google Translate can be useful for this.

Please try to reformulate the problem in terms of what are you trying to achieve, what you've tried and how does the outcome differs from the expectations. Knowing that will help me to help you ;-)

**Comment by questionpython** [comment]
i want use from `go tool yacc`, but say not exist!

**Comment by cznic** [comment]
Ah, here we go ;-)

If you want to use `go tool yacc` command you need to do

```bash
go get -u golang.org/x/tools/cmd/goyacc
```

**Comment by questionpython** [comment]
$ go get -u golang.org/x/tools/cmd/goyacc
------


$ go tool yacc
---------
go tool: no such tool "yacc"

**Comment by cznic** [comment]
I've raised a question at the mailing list: https://groups.google.com/d/topic/golang-nuts/oiaNlQdqPb8/discussion

Sorry for providing you with a non-working recipe, I am surprised that it does not work.

**Comment by questionpython** [comment]
tank you...., i waiting for you. :+1: 
again tank you from you.

**Comment by cznic** [comment]
Update: As far as I can tell now, the problem is that the `go` tool no longer supports the `yacc` subcommand at all.

After doing

```bash
$ go get -u golang.org/x/tools/cmd/goyacc
```

on must invoke the tool directly

```bash
$ goyacc
```

Note that the `golang.org/...` and `cznic/goyacc` repositories both have a and via `go get` both install command `goyacc`. They are to a certain extent compatible, which one you should choose to install depends on what features you need to use.
